### ISSUE TITLE
pylint . is inconsistent with pylint ../$(basename $(pwd))

### ISSUE THREAD

**Issue by tom-d** [PyCQA/pylint#2593]
### Steps to reproduce

```bash
mkdir tmp
touch __init__.py
mkdir tmp/lib
echo 'a = b' > tmp/lib/__init__.py
cd tmp
```

### Current behavior

``pylint . && echo $?``
0

``pylint ../tmp``
E0602: Undefined variable 'b' (undefined-variable)

``(cd .. && pylint tmp)``
E0602: Undefined variable 'b' (undefined-variable)

### Expected behavior
 
pylint .
E0602: Undefined variable 'b' (undefined-variable)

### pylint --version output

pylint 2.1.1
astroid 2.1.0-dev
Python 3.6.2 (v3.6.2:5fd33b5926, Jul 16 2017, 20:11:06) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]

**Comment by tom-d** [comment]
correction to the above steps to reproduce..

```
mkdir tmp
touch tmp/__init__.py
mkdir tmp/lib
echo 'a = b' > tmp/lib/__init__.py
cd tmp
```

**Comment by PCManticore** [comment]
Thanks for creating an issue @tom-d !

**Comment by Pierre-Sassoulas** [comment]
This is fixed by using the ``--recursive=y`` option, and will be definitely fixed with #5701
